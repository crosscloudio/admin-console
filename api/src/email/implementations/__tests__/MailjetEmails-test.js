jest.mock('node-fetch');

describe('MailjetEmails', () => {
  beforeEach(() => {
    Date.now = jest.fn(
      // Tue May 30 2017 10:40:16 GMT+0200 (CEST)
      () => 1496133616539
    );
  });

  describe('#sendResetPasswordEmail', () => {
    it('should sent email', async () => {
      jest.resetModules();
      process.env.MAILJET_API_KEY = 'sample key';
      process.env.MAILJET_API_SECRET = 'sample secret';
      process.env.MAILJET_SENDER_EMAIL = 'me@example.com';
      delete process.env.SMTP_SENDER_EMAIL;
      delete process.env.SMTP_URL;

      const fetch = require('node-fetch'); // eslint-disable-line global-require
      const MailjetEmails = require(// eslint-disable-line global-require
      '../MailjetEmails').default;

      const helper = new MailjetEmails();
      const fakeUser = {
        id: 1,
        name: 'John Smith',
        email: 'sample@test.org',
      };

      // `Set password` email
      await helper.sendResetPasswordEmail(fakeUser, { initial: true });
      expect(fetch.getMockArgs()).toMatchSnapshot();

      // `Reset password` email
      await helper.sendResetPasswordEmail(fakeUser);
      expect(fetch.getMockArgs()).toMatchSnapshot();
    });
  });
});
