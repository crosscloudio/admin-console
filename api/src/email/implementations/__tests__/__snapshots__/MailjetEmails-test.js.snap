// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailjetEmails #sendResetPasswordEmail should sent email 1`] = `
Array [
  "https://api.mailjet.com/v3/send",
  Object {
    "body": "{\\"FromEmail\\":\\"me@example.com\\",\\"FromName\\":\\"CrossCloud\\",\\"Recipients\\":[{\\"Email\\":\\"sample@test.org\\",\\"Name\\":\\"John Smith\\"}],\\"Subject\\":\\"Set CrossCloud password\\",\\"Text-part\\":\\"Hi John Smith,\\\\n\\\\nSet your password here: http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w\\\\n\\\\nThanks,\\\\nCrossCloud team\\\\n\\",\\"Html-part\\":\\"<p>Hi John Smith,</p>\\\\n\\\\n<p>Click the link below to set your password:</p>\\\\n<p><a href=\\\\\\"http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w\\\\\\">http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w</a></p>\\\\n\\\\n<p>Thanks,<br>\\\\nCrossCloud team</p>\\\\n\\",\\"Mj-trackclick\\":1}",
    "headers": Object {
      "Authorization": "Basic c2FtcGxlIGtleTpzYW1wbGUgc2VjcmV0",
      "Content-Type": "application/json",
    },
    "method": "POST",
  },
]
`;

exports[`MailjetEmails #sendResetPasswordEmail should sent email 2`] = `
Array [
  "https://api.mailjet.com/v3/send",
  Object {
    "body": "{\\"FromEmail\\":\\"me@example.com\\",\\"FromName\\":\\"CrossCloud\\",\\"Recipients\\":[{\\"Email\\":\\"sample@test.org\\",\\"Name\\":\\"John Smith\\"}],\\"Subject\\":\\"Reset CrossCloud password\\",\\"Text-part\\":\\"Hi John Smith,\\\\n\\\\nReset your password here: http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w\\\\n\\\\nThanks,\\\\nCrossCloud team\\\\n\\",\\"Html-part\\":\\"<p>Hi John Smith,</p>\\\\n\\\\n<p>Click the link below to reset your password:</p>\\\\n<p><a href=\\\\\\"http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w\\\\\\">http://localhost:8000/reset-password/eyJhbGciOiJIUzUxMiIsInR5cCI6IkpXVCJ9.eyJ1aWQiOjEsImlhdCI6MTQ5NjEzMzYxNiwiZXhwIjoxNDk2MzkyODE2LCJzdWIiOiJwYXNzd29yZC1zZXQifQ.wvewANIKzbgJ_HGXo6zRRlNXXFjCC7GZ7E7KRd1zJR4_CzRKMtUVJWkvJNmMKbTueRN9iCpn2bQFSEYeol-d2w</a></p>\\\\n\\\\n<p>Thanks,<br>\\\\nCrossCloud team</p>\\\\n\\",\\"Mj-trackclick\\":1}",
    "headers": Object {
      "Authorization": "Basic c2FtcGxlIGtleTpzYW1wbGUgc2VjcmV0",
      "Content-Type": "application/json",
    },
    "method": "POST",
  },
]
`;
